src/course-libraries/data/apiHooks.ts

```diff
@@ -1,20 +1,94 @@
 import {
+  useInfiniteQuery,
   useQuery,
 } from '@tanstack/react-query';
-import { getEntityLinksByDownstreamContext } from './api';
+import { getEntityLinks, getEntityLinksSummaryByDownstreamContext, getUnpaginatedEntityLinks } from './api';
 
 export const courseLibrariesQueryKeys = {
   all: ['courseLibraries'],
-  courseLibraries: (courseKey?: string) => [...courseLibrariesQueryKeys.all, courseKey],
+  courseLibraries: (courseId?: string) => [...courseLibrariesQueryKeys.all, courseId],
+  courseReadyToSyncLibraries: ({ courseId, readyToSync, upstreamUsageKey }: {
+    courseId?: string,
+    readyToSync?: boolean,
+    upstreamUsageKey?: string,
+    pageSize?: number,
+  }) => {
+    const key: Array<string | boolean | number> = [...courseLibrariesQueryKeys.all];
+    if (courseId !== undefined) {
+      key.push(courseId);
+    }
+    if (readyToSync !== undefined) {
+      key.push(readyToSync);
+    }
+    if (upstreamUsageKey !== undefined) {
+      key.push(upstreamUsageKey);
+    }
+    return key;
+  },
+  courseLibrariesSummary: (courseId?: string) => [...courseLibrariesQueryKeys.courseLibraries(courseId), 'summary'],
 };
 
 /**
- * Hook to fetch a content library by its ID.
+ * Hook to fetch publishable entity links by course key.
```

Comment:
```suggestion
 * Hook to fetch publishable entity links by course key.
 * (That is, get a list of the library components used in the given course.)
```
People not super familiar with this part of the code may not know what "publishable entity links" are.